app/src/main/java/org/oppia/android/app/devoptions/featureflags/FeatureFlagsFragmentPresenter.kt

```diff
@@ -84,7 +93,24 @@ class FeatureFlagsFragmentPresenter @Inject constructor(
       model.isChecked.set(featureFlagStates[model.featureFlagId])
     }
     model.onFeatureFlagToggleCallback = { id, value ->
-      featureFlagStates[id] = value
+      if (model.currentValue == value)
+        featureFlagStates.remove(id)
+      else
+        featureFlagStates[id] = value
```

Comment:
I remember it was suggested somewhere to keep it this way as preferred for readability and consistency with codebase.
```suggestion
      if (model.currentValue == value) {
        featureFlagStates.remove(id)
      } else {
        featureFlagStates[id] = value
      }
```

Reply:
updated